rav1e fails to build correctly

09:36:02   Running git update for rav1e...
09:36:03 ┌ rav1e git  ................................ [Recently updated]
09:36:03 ├ Running rust.update...
09:36:07 ├ Running rust.build...
09:37:49 ├ Running install-rav1e-c...
Likely error (tail of the failed operation logfile):
error: linking with `x86_64-w64-mingw32-gcc` failed: exit code: 1
  |
  = note: "x86_64-w64-mingw32-gcc" "-fno-use-linker-plugin" "-Wl,--dynamicbase" "-Wl,--disable-auto-image-base" "-m64" "-Wl,--high-entropy-va" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsbegin.o" "C:\\MABS\\msys64\\tmp\\rustcDi2a27\\symbols.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.19y8w66989r7qciu.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.1a5by3y2a9x1e6sj.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.1cy3894p0lhqkxq6.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.1fmam6tutuusr7po.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.1hckezmcby53y2ey.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.1hwggq5mngcc4yag.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.1vewi0pj1ahg8kjo.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.22rnjimfqmvfx78v.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.2924v10qqb8aq74q.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.2bq67sd7h2zku4j3.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.2d5tmbuwwicx2yu1.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.2pjxs0m0bp5sbhv9.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.2ve0j1e1h4ok261v.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.37grw5fdg6n2xen6.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.3cp3490wq0lbz80i.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.3e78b09wt9vwr8wu.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.3esoz838fw183m42.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.3j3amcf300a30mdq.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.3jpkxeelm28jlut2.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.3k78xecyptfen6ni.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.3mtooct846rb0vph.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.3nrj424sv0ckt22i.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.3tty2at39grvjgq2.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.3udugn9ov662q3al.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.3yd5b6wvoxyjavcq.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.4rqm5ekcvd3zxtek.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.4ujagt127hjxklnq.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.51obq6sigymgfdzy.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.538pwfy3bx4wqt4h.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.5e8f6ye5wqgmerbr.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.5frbdpsd1w4d4us1.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.qsy4oiqn3d1ir0f.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.rbepo6i0a09xefp.rcgu.o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.576w59eu3mhkcx09.rcgu.o" "-L" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps" "-L" "C:/MABS/msys64/mingw64/lib" "-L" "C:/MABS/msys64/mingw64/lib" "-L" "/mingw64/lib" "-L" "C:/MABS/msys64/mingw64/lib" "-L" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\libz-sys-180cc2788a7480ca\\out\\lib" "-L" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\libz-sys-180cc2788a7480ca\\out\\lib" "-L" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-Wl,-Bstatic" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libbuilt-433e493a417e21a0.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libgit2-f05055654d5b41f4.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\liburl-4c2c6dfbb206fc9f.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libidna-fb7348a880075a2a.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libunicode_normalization-0ce9d408b8a9e609.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libtinyvec-2dc4bc00fb6b903f.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libtinyvec_macros-5b5adbdddc27331e.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libunicode_bidi-2bb3a22f98e6d8ab.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libform_urlencoded-8ffe1db1fa534a06.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libpercent_encoding-254212247873a083.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\liblog-935ed9b25e1d8a3f.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\liblibgit2_sys-013e3bade58aad4e.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\liblibz_sys-40924b80b21eeb75.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\liblibc-a89a6fbc8d16b51d.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libbitflags-652ac28215865aca.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libcc-e65e1080551df1e1.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libjobserver-376909902d264910.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libonce_cell-5fd503e788208213.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libnasm_rs-43b998ccb52233a4.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\librayon-e33613f67d25731a.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\librayon_core-0abf335c4bd20967.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libcrossbeam_deque-8159bbb23a68747a.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libcrossbeam_epoch-4d1a2d19ed0d3fc9.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libcrossbeam_utils-0e917f4971924dba.rlib" "C:\\MABS\\build\\rav1e-git\\target\\release\\deps\\libeither-3399431130829c7d.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libstd-7e3d0778af0bb50b.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libpanic_unwind-125340f7b648b1da.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libobject-585dac36643d4405.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libmemchr-0f644536f213afef.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libaddr2line-d80034aa65d1a3f8.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libgimli-975cce95f29c904a.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\librustc_demangle-a4a7f3a04e1d729b.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libstd_detect-851ee9869c1cab83.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libhashbrown-94a4f7b41d9c1eed.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\librustc_std_workspace_alloc-96f56a68bc789848.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libminiz_oxide-511a046dd37edb1f.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libadler-5618743c182f257a.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libunwind-9a552f87bb923250.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcfg_if-5acec5c68c7475b1.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\liblibc-3667c04cdc2f1315.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\liballoc-7a2990ff4cc1897e.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\librustc_std_workspace_core-b410c266ca01faf5.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcore-a8e2bbea32261e7b.rlib" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcompiler_builtins-1bc3f2958faab565.rlib" "-Wl,-Bdynamic" "-lgit2" "-lhttp_parser" "-lssh2" "-lsecur32" "-liconv" "-lssl" "-lcrypto" "-lws2_32" "-lgdi32" "-lcrypt32" "-lpcre2-8" "-lz" "-ladvapi32" "-lkernel32" "-lole32" "-loleaut32" "-lkernel32" "-ladvapi32" "-lkernel32" "-lntdll" "-luserenv" "-lws2_32" "-lkernel32" "-lws2_32" "-lkernel32" "-lgcc_eh" "-l:libpthread.a" "-lmsvcrt" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt" "-luser32" "-lkernel32" "-Wl,--nxcompat" "-L" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-o" "C:\\MABS\\build\\rav1e-git\\target\\release\\build\\rav1e-7781da8901b4306a\\build_script_build-7781da8901b4306a.exe" "-Wl,--gc-sections" "-no-pie" "-nodefaultlibs" "C:\\MABS\\msys64\\mingw64\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsend.o"
  = note: C:/MABS/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lgit2: No such file or directory
          collect2.exe: error: ld returned 1 exit status


error: could not compile `rav1e` (build script) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
Error: CliError { error: Some(1 job failed), exit_code: 101 }
install-rav1e-c failed. Check C:/MABS/build/rav1e-git/ab-suite.install-rav1e-c.log
This is required for other packages, so this script will exit.
09:38:11   Creating diagnostics file...

Attach C:\MABS\build\logs.zip to the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.

Try running the build again at a later time.


[logs.zip](https://github.com/m-ab-s/media-autobuild_suite/files/15404087/logs.zip)

--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -1144,8 +1144,8 @@ if { [[ $rav1e = y ]] || [[ $libavif = y ]] || enabled librav1e; } &&
     do_uninstall "${_check[@]}" include/rav1e
 
     # We want to hide libgit2 unless we have a static library
-    _libgit2_pc="$MINGW_PACKAGE_PREFIX/lib/pkgconfig/libgit2.pc"
-    if ! [[ -f $MINGW_PACKAGE_PREFIX/lib/libgit2.a ]]; then
+    _libgit2_pc="$MINGW_PREFIX/lib/pkgconfig/libgit2.pc"
+    if ! [[ -f $MINGW_PREFIX/lib/libgit2.a ]]; then
         if  [[ -f $_libgit2_pc ]]; then
             mv -f "$_libgit2_pc"{,.dyn}
         fi

